newroutes/users.ts

```diff
@@ -1,20 +1,34 @@
-const autoBind = require("auto-bind")
-const express = require("express")
-const validator = require("validator")
+import autoBind from "auto-bind"
+import express, { Request, Response } from "express"
+import validator from "validator"
 
-const logger = require("@logger/logger")
+import logger from "@logger/logger"
 
-const { BadRequestError } = require("@errors/BadRequestError")
+import { BadRequestError } from "@errors/BadRequestError"
 
-const { attachReadRouteHandlerWrapper } = require("@middleware/routeHandler")
+import { attachReadRouteHandlerWrapper } from "@middleware/routeHandler"
 
-class UsersRouter {
-  constructor({ usersService }) {
+import UsersService from "@services/identity/UsersService"
+
+import { isError, RequestHandler } from "../types"
+
+interface UsersRouterProps {
+  usersService: UsersService
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export class UsersRouter {
+  private readonly usersService
+
+  constructor({ usersService }: UsersRouterProps) {
     this.usersService = usersService
     autoBind(this)
   }
 
-  async sendEmailOtp(req, res) {
+  sendEmailOtp: RequestHandler<never, unknown, { email?: string }> = async (
+    req,
+    res
+  ) => {
```

Comment:
Why do we specify never/unknown? Do the remaining 2 unspecified params remain as unknown - we could possibly explicitly define the remaining 2 as unknown if so?

Reply:
to clarify this - it's because the first two need to be specified in order to get to specify the concrete type. i specified the first as `never` because it's for the params, which is not set. 

res body is the second param and it's not used but _could be,_ hence the `unknown`.

Reply:
thanks for the explanation - and for the last 2 params, if we don't specify them, will they remain `unknown` by default?

Reply:
since this `RequestHandler` is internal, yes! but the express one defaults it to `any` (hence the need for an internal one)